### ISSUE TITLE
Elixir installer on Windows 10 installs Elixir and IEx as broken

### ISSUE THREAD

**Issue by bmitchell-cfs** [elixir-lang/elixir#10110]
### Environment

* Elixir & Erlang/OTP versions (elixir --version):

    * Elixir does not run, but Elixir 1.10.3 was selected to be installed.
![image](https://user-images.githubusercontent.com/64422988/85103254-50807380-b1d4-11ea-9555-a93cab144a98.png)

    * The Erlang version, as reported by the installer is 9.2.
![image](https://user-images.githubusercontent.com/64422988/85103300-655d0700-b1d4-11ea-865b-4fc5e9772653.png)

* Operating system: Windows 10, version 1909, OS build 18363.900

### Current behavior

The Windows installer found [here](https://elixir-lang.org/install.html#windows) does not properly install Elixir. The installer reports no errors, but after installation, both iex.bat and elixir.bat fail by crashing. The application werl.exe fails similarly but then automatically closes without the user being able to inspect the crash.

When running `iex.bat` in Windows Terminal with PowerShell, I get:

```
> iex.bat
2020-06-19 02:16:39 Loading of ~ts failed: ~p

        "c:/Program Files (x86)/Elixir/lib/iex/ebin/Elixir.IEx.CLI.beam"
        badfile
2020-06-19 02:16:39 ~s~n
        "beam/beam_load.c(1862): Error loading module 'Elixir.IEx.CLI':\n  This BEAM file was compiled for a later version of the run-time system than 20.\n  To fix this, please recompile this module with an 20 compiler.\n  (Use of opcode 162; this emulator supports only up to 159.)\n"
2020-06-19 02:16:39 crash_report
    initial_call: {supervisor_bridge,user_sup,['Argument__1']}
    pid: <0.49.0>
    registered_name: []
    error_info: {error,undef,[{'Elixir.IEx.CLI',start,[],[]},{user_sup,start_user,3,[{file,"user_sup.erl"},{line,100}]},{user_sup,init,1,[{file,"user_sup.erl"},{line,49}]},{supervisor_bridge,init,1,[{file,"supervisor_bridge.erl"},{line,80}]},{gen_server,init_it,2,[{file,"gen_server.erl"},{line,365}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,333}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,247}]}]}
    ancestors: [kernel_sup,<0.36.0>]
    message_queue_len: 0
    messages: []
    links: [<0.37.0>]
    dictionary: []
    trap_exit: true
    status: running
    heap_size: 987
    stack_size: 27
    reductions: 223
2020-06-19 02:16:39 supervisor_report
    supervisor: {local,kernel_sup}
    errorContext: start_error
    reason: {undef,[{'Elixir.IEx.CLI',start,[],[]},{user_sup,start_user,3,[{file,"user_sup.erl"},{line,100}]},{user_sup,init,1,[{file,"user_sup.erl"},{line,49}]},{supervisor_bridge,init,1,[{file,"supervisor_bridge.erl"},{line,80}]},{gen_server,init_it,2,[{file,"gen_server.erl"},{line,365}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,333}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,247}]}]}
    offender: [{pid,undefined},{id,user},{mfargs,{user_sup,start,[]}},{restart_type,temporary},{shutdown,2000},{child_type,supervisor}]
2020-06-19 02:16:39 crash_report
    initial_call: {application_master,init,['Argument__1','Argument__2','Argument__3','Argument__4']}
    pid: <0.35.0>
    registered_name: []
    error_info: {exit,{{shutdown,{failed_to_start_child,user,{undef,[{'Elixir.IEx.CLI',start,[],[]},{user_sup,start_user,3,[{file,"user_sup.erl"},{line,100}]},{user_sup,init,1,[{file,"user_sup.erl"},{line,49}]},{supervisor_bridge,init,1,[{file,"supervisor_bridge.erl"},{line,80}]},{gen_server,init_it,2,[{file,"gen_server.erl"},{line,365}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,333}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,247}]}]}}},{kernel,start,[normal,[]]}},[{application_master,init,4,[{file,"application_master.erl"},{line,134}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,247}]}]}
    ancestors: [<0.34.0>]
    message_queue_len: 1
    messages: [{'EXIT',<0.36.0>,normal}]
    links: [<0.34.0>,<0.33.0>]
    dictionary: []
    trap_exit: true
    status: running
    heap_size: 987
    stack_size: 27
    reductions: 243
2020-06-19 02:16:39 std_info
    application: kernel
    exited: {{shutdown,{failed_to_start_child,user,{undef,[{'Elixir.IEx.CLI',start,[],[]},{user_sup,start_user,3,[{file,"user_sup.erl"},{line,100}]},{user_sup,init,1,[{file,"user_sup.erl"},{line,49}]},{supervisor_bridge,init,1,[{file,"supervisor_bridge.erl"},{line,80}]},{gen_server,init_it,2,[{file,"gen_server.erl"},{line,365}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,333}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,247}]}]}}},{kernel,start,[normal,[]]}}
    type: permanent
{"Kernel pid terminated",application_controller,"{application_start_failure,kernel,{{shutdown,{failed_to_start_child,user,{undef,[{'Elixir.IEx.CLI',start,[],[]},{user_sup,start_user,3,[{file,\"user_sup.erl\"},{line,100}]},{user_sup,init,1,[{file,\"user_sup.erl\"},{line,49}]},{supervisor_bridge,init,1,[{file,\"supervisor_bridge.erl\"},{line,80}]},{gen_server,init_it,2,[{file,\"gen_server.erl\"},{line,365}]},{gen_server,init_it,6,[{file,\"gen_server.erl\"},{line,333}]},{proc_lib,init_p_do_apply,3,[{file,\"proc_lib.erl\"},{line,247}]}]}}},{kernel,start,[normal,[]]}}}"}
Kernel pid terminated (application_controller) ({application_start_failure,kernel,{{shutdown,{failed_to_start_child,user,{undef,[{'Elixir.IEx.CLI',start,[],[]},{user_sup,start_user,3,[{file,"user_sup.

Crash dump is being written to: erl_crash.dump...done
```

When running `elixir.bat --version` in Windows Terminal with PowerShell, I get:

````
> elixir.bat --version
{"init terminating in do_boot",{undef,[{elixir,start_cli,[],[]},{init,start_em,1,[{file,"init.erl"},{line,1085}]},{init,do_boot,3,[{file,"init.erl"},{line,793}]}]}}
init terminating in do_boot ({undef,[{elixir,start_cli,[],[]},{init,start_em,1,[{_},{_}]},{init,do_boot,3,[{_},{_}]}]})

Crash dump is being written to: erl_crash.dump...done
````

### Expected behavior

The expectation is that the installer should just work. The crash message seems to imply a version mismatch with Erlang, but the installer should properly install both Erlang and Elixir with the correct versions.

**Comment by josevalim** [comment]
Hi @bmitchell-cfs! It seems there was already an Erlang installed in the system but it was an old installer. Can you please run the installer again and then make sure that you install the most recent Erlang version too? You can also pick the most recent Erlang from here too: https://www.erlang.org/downloads

**Comment by bmitchell-cfs** [comment]
Thanks for the information. I didn't install Erlang myself. Looking into this further, it seems Erlang 9.2 was likely installed by some National Instruments software, which seem to use RabbitMQ. Reading the language more closely, I see that the Elixir installer was choosing to use the already installed Erlang 9.2 rather than doing the install itself, which was what I originally assumed.

Should this be transitioned to a bug or feature instead of being closed? I think closing this without understanding the complete issue is premature. Because I think that the installer should automatically install a compatible version of Erlang. If it can't, then it should warn that the installed version is not compatible and not allow configuration with the incompatible version. The expectation on Windows is that installers should not leave the installed software broken and should instead handle the issue automatically or warn. So I think the issue potentially transitions to "the Elixir installer should not allow configuration with an incompatible Erlang version" with a proposed fix of automatically installing a compatible version or warning of the incompatibility and not continuing.

Some additional information is that even after uninstalling Elixir, then manually installing the latest version of Erlang (Erlang/OTP 23.0), and then re-installing Elixir, the Elixir installer still automatically suggests configuring with the earlier, incompatible version, even though there is a later, compatible version installed.

![image](https://user-images.githubusercontent.com/64422988/85109070-bffb6080-b1de-11ea-94ea-b7a090d52d5b.png)

And even after pointing Elixir to the correct version of Erlang, in this case 23.0, the installer does not successfully install Elixir, as I still get crashes. It appears the crashes are the same as above, so it seems that even after pointing the installer to Erlang/OTP 23.0, Elixir is still using Erlang 9.2. This is path related, because when I type `erl` into a terminal, I get the Erlang 9.2 version. The Elixir installer explicitly asked me if I wanted it to add Erlang/OTP 23.0 to my path. (I didn't take a screenshot of this but confirmed it did add it to my system path.) The Elixir tooling is just using whatever Erlang is returned first in the path, which is a separate issue, in my opinion. It should use the version I pointed it to during the installation process and not use whatever Erlang comes first in the system or user path, at least that is the expectation. I confirmed this is the case by moving erl-23.0 above (earlier than) erl9.2 in the system path, and with no other changes, iex.bat now runs successfully.

I would summarize the outstanding issues as:
1. The Elixir installer should automatically install a compatible version of Erlang if one is not installed.
2. The Elixir installer should not allow configuration with an incompatible Erlang version.
3. The Elixir installer should automatically recognize the latest installed version of Erlang.
4. The Elixir Installer and/or Elixir tools, should properly honor the requested Erlang version configuration and should not use whatever Erlang is returned first in the path.
